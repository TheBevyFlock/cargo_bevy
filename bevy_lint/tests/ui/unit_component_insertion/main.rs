#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::unit_component_insertion)]
#![allow(unused_variables)]

use bevy::prelude::*;
use std::f32::consts::PI;

fn main() {
    App::new().add_systems(Startup, test);
}

fn test(mut commands: Commands) {
    commands.spawn((
        Name::new("Decal"),
        Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
        //~^ ERROR: Expression returns `unit` and results in an empty component insertion
        example()
        //~^ ERROR: Expression returns `unit` and results in an empty component insertion
    ));
}

fn example() {
    let foo = 1;
}
