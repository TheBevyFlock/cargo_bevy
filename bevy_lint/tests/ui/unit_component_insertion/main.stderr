error: Expression returns `unit` and results in an empty component insertion
  --> tests/ui/unit_component_insertion/main.rs:16:64
   |
16 |         Transform::from_translation(Vec3::new(0.75, 0.0, 0.0)).rotate_z(PI / 4.0),
   |                                                                ^^^^^^^^
   |
note: the lint level is defined here
  --> tests/ui/unit_component_insertion/main.rs:3:9
   |
3  | #![deny(bevy::unit_component_insertion)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Expression returns `unit` and results in an empty component insertion
  --> tests/ui/unit_component_insertion/main.rs:18:9
   |
18 |         example()
   |         ^^^^^^^

error: aborting due to 2 previous errors

